python/subprojects/arrow-nanoarrow.wrap

```diff
@@ -14,12 +14,14 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-#
-# This file is part of 'miniver': https://github.com/jbweston/miniver
 
-# Replaced by version-bumping scripts at release time
-version = "0.6.0.dev0"
+[wrap-file]
+directory = arrow-nanoarrow-b78c0395a6fe74e776daf07933583f06567fb99c
```

Comment:
This is just pointing to a single commit here that includes some of the changes made to the top level `meson.build` configuration. In a follow up ideally point to main or even a wrapdb release

Reply:
The commit level integration is managed through the symlink to the subprojects folder. A wrap file doesn't work like that - instead it just copies the library at a point in time to subprojects, and you would have to manually run `meson wrap update` constantly to get that

Reply:
To try and bring that all together...I ideally wanted to have a subprojects folder that looked like:

```
python/
  subprojects/
    arrow-nanoarrow/  -- this is a symlink to the project root
    arrow-nanoarrow.wrap
```

Ideally, if meson finds the `arrow-nanoarrow` subproject it should not need the wrap file at all, and just happily use the symlink for a live integration. When it comes to sdists where the symlink will not exist, the wrap file should be responsible for resolving and unpacking a particular `arrow-nanoarrow` distribution to the subprojects folder (probably from the wrap db, but in this PR I have it pointing to this branch to get the config updates)

Due to the "bug" I linked elsewhere in this PR, Meson confusingly still downloads the contents of `arrow-nanoarrow.wrap` even when the `arrow-nanoarrow` symlink exists, so to work around that the local directory structure instead looks like:

```
python/
  subprojects/
    arrow-nanoarrow-dev/  -- this is a symlink to the project root
    arrow-nanoarrow.wrap
```

And the configuration looks like:

```python
nanoarrow_proj = subproject('arrow-nanoarrow-dev', required: false)
if not nanoarrow_proj.found()
  nanoarrow_proj = subproject('arrow-nanoarrow')
endif
```

When developing locally, the arrow-nanoarrow-dev symlink will always be prioritized. Of course, when we create source distributions we end up removing symlinks, so the wrap file comes into play in those instances.

Hope that makes sense but let me know if anything is unclear

Reply:
Ah OK. In that case we can probably get rid of the wrap file then; just need to figure out the right way/hook to install a copy into subprojects when generating the sdist

Reply:
Meson-python and meson both actually have a dist option to include subprojects, which I think is where our best bet is

https://meson-python.readthedocs.io/en/stable/how-to-guides/meson-args.html

Of course, it doesn't seem to work right now (I think the symlink behavior tricks it into failure) but let me research that a bit more. Would rather let the build system take care of this if it can

Reply:
[`meson.add_dist_script`](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonadd_dist_script) may be useful here? It will run when building an sdist, and you can invoke a Python script to copy exactly the files/dirs you want under `subprojects/`.

Reply:
Ah OK I think that's a great idea. The `--include-subprojects` flag I was looking at seems to just copy the symlink, but not deeply copy the subproject. So a custom script is probably the safest bet